src/main/java/software/aws/neptune/opencypher/OpenCypherIAMRequestGenerator.java

```diff
@@ -76,4 +82,64 @@ private static String getAuthInfoJson(final Request<Void> request) {
 
         return GSON.toJson(obj);
     }
+
+    /**
+     * This class is derived from the AWS documentation on accessing Amazon Neptune with IAM authentication.
+     * For more details and information, please refer to:
+     * <a href="https://docs.aws.amazon.com/neptune/latest/userguide/access-graph-opencypher-bolt.html#access-graph-opencypher-bolt-java-iam-auth">...</a>
+     */
+
+    public static class NeptuneAuthToken extends InternalAuthToken {
+        private static final String SCHEME = "basic";
+        private static final String REALM = "realm";
+        private static final String SERVICE_NAME = "neptune-db";
+        private static final String DUMMY_USERNAME = "username";
+        @NonNull
+        private final String region;
+        @NonNull
+        @Getter
+        private final String url;
+        @NonNull
+        private final AWSCredentialsProvider credentialsProvider;
+        private final Gson gson = new Gson();
+
+        @Builder
+        private NeptuneAuthToken(
+                @NonNull final String region,
+                @NonNull final String url,
+                @NonNull final AWSCredentialsProvider credentialsProvider
+        ) {
+            // The superclass caches the result of toMap(), which we don't want
+            super(Collections.emptyMap());
+            this.region = region;
+            this.url = url;
+            this.credentialsProvider = credentialsProvider;
+        }
+
+        @Override
+        public Map<String, Value> toMap() {
+            final Map<String, Value> map = new HashMap<>();
+            map.put(SCHEME_KEY, Values.value(SCHEME));
+            map.put(PRINCIPAL_KEY, Values.value(DUMMY_USERNAME));
+            map.put(CREDENTIALS_KEY, new StringValue(getSignedHeader()));
+            map.put(REALM_KEY, Values.value(REALM));
+
+            return map;
+        }
+
+        private String getSignedHeader() {
```

Comment:
Actually just realized that this method is duplicated above, we might want to refactor this class and the way we are calling it in the driver.

Reply:
just remove the above one since it was private anyhow